File: dataset/human/python/file_3669.py
from collections import Counter

def Eratosthenes(n):
    is_prime = [True] * (n + 1)
    is_prime[0] = False
    is_prime[1] = False
    for i in range(2, int(n ** 0.5) + 1):
        if not is_prime[i]:
            continue
        for j in range(i * 2, n + 1, i):
            is_prime[j] = False
    return [i for i in range(n + 1) if is_prime[i]]

n = int(input())
n2 = n

if n == 1:
    print(0)
    exit()

prime_factors = [] #素因数
for i in Eratosthenes(int(n ** 0.5)):
    if i**2 > n2:
        break
    while n2 % i == 0:
        prime_factors.append(i)
        n2 //= i
if n2 > 1:
    prime_factors.append(n2)

#↑までで素因数分解

cntr = Counter(prime_factors)
cnt = 0
#for k, v in cntr.items():
for v in cntr.values():
    sisuu = v
    sub = 1
    while sisuu - sub >= 0:
        sisuu -= sub
        cnt += 1
        sub += 1
print(cnt)
"""
ans = []
list = []
for i in prime_factors:
    for j in list:
        if j % i == 0:
            tmp = list.pop(list.index(j)) * i
            if tmp in ans:
                list.append(tmp)
                continue
            ans.append(tmp)
    if i in list:
        tmp = list.pop(list.index(i)) * i
        if tmp in ans:
            list.append(tmp)
            continue
        ans.append(tmp)
    elif i in ans:
        list.append(i)
    else:
        ans.append(i)

print(len(ans))
"""

File: dataset/human/cpp/file_3669.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main(){
  long long cnt=0,n,ans=0;
  cin>>n;
  vector<int>a(n);
  vector<int>b(n);
  for(int i=0;i<n;i++){
    cin>>a[i];
  }
  for(int i=0;i<n;i++){
    if(a[i]<0){
      a[i]=(-1)*a[i];
      ans+=a[i];
      cnt++;
    }
    else{
      ans+=a[i];
     }
  } 
  for(int i=0;i<n;i++){
    if(a[i]<0){
      b[i]=a[i]*(-1);
  }
    else{
      b[i]=a[i];
    }
  }
  sort(b.begin(),b.end());
  if(cnt%2==0){
    cout<<ans<<endl;
    return 0;
  }
  else{
    cout<<ans-b[0]*2<<endl;
    return 0;
  }
}
